back-end/routes/users.js

```diff
@@ -91,21 +81,31 @@ router.post('/', [
       // get email password from request body
       const { email, password } = req.body;
 
+      // check if email is already in use
+      let dbResults = await getUserByEmail(email);
+      if(dbResults) {
+        return res.status(400).send('Email address is already in use.');
+      }
+
       // salt and hash password
       const salt = await bcrypt.genSalt();
       const hashedPassword = await bcrypt.hash(password, salt);
 
       // create user in DB
       const createUserQuery = `INSERT INTO users(email, password, created_at) \
-VALUES('${email}', '${hashedPassword}', CURRENT_TIMESTAMP) returning *`
+VALUES('${email}', '${hashedPassword}', CURRENT_TIMESTAMP) RETURNING *`
+      dbResults = await getRowFromDb(createUserQuery);
 
-      let dbResults = await getRowFromDb(createUserQuery);
+      let responseBody = await getFullUserProfile(dbResults.id);
```

Comment:
Probably overkill, since the new user won't have a connected github_profile yet, but why not 🤷